### PR TITLE
[Bug][Array] cast array element to same type

### LINKED ISSUE #10978: [Bug][Array] wrong result when select array contain float and integer
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```sql
MySQL [regression_test_rollup]> SELECT [1.0, 2];
+-------------------------------------+
| ARRAY(1.0, 2)                       |
+-------------------------------------+
| [1, -7564368093082427205.321810946] |
+-------------------------------------+
1 row in set (0.004 sec)

MySQL [regression_test_rollup]> SELECT [1.0, 2];
+----------------------------+
| ARRAY(1.0, 2)              |
+----------------------------+
| [1, 18446744073.709551618] |
+----------------------------+
1 row in set (0.005 sec)

MySQL [regression_test_rollup]> SELECT [1.0, 2];
+-------------------------------------+
| ARRAY(1.0, 2)                       |
+-------------------------------------+
| [1, -2290455768325073572.223176706] |
+-------------------------------------+
1 row in set (0.004 sec)
```

### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #10978

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ArrayLiteral.java [hunk 1]
```diff
@@ -39,20 +38,28 @@ public ArrayLiteral() {
...
 
-    public ArrayLiteral(LiteralExpr... v) {
+    public ArrayLiteral(LiteralExpr... exprs) throws AnalysisException {
         Type itemType = Type.NULL;
         boolean containsNull = false;
-        for (LiteralExpr expr : v) {
-            if (itemType == Type.NULL || expr.type.getSlotSize() > itemType.getSlotSize()) {
-                itemType = expr.type;
+        for (LiteralExpr expr : exprs) {
+            if (itemType == Type.NULL || expr.getType().getSlotSize() > itemType.getSlotSize()) {
```
Comment (cambyzju) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ArrayLiteral.java:45]: use `getAssignmentCompatibleType` is better than use `getSlotSize`

**Comment by cambyzju** [comment]
LGTM

**Comment by cambyzju** [comment]
FE UT need to fix

**Comment by eldenmoon** [comment]
LGTM

**Review by dataroaring — APPROVED** [review]

**Pull request merged**